pkgs/development/libraries/wxwidgets/wxGTK32.nix

```diff
@@ -0,0 +1,147 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchurl
+, gnome2
+, gst_all_1
+, gtk2
+, gtk3
+, libGL
+, libGLU
+, libSM
+, libXinerama
+, libXtst
+, libXxf86vm
+, pkg-config
+, xorgproto
+, compat28 ? false
+, compat30 ? true
+, unicode ? true
+, withGtk2 ? (!stdenv.isDarwin)
+, withMesa ? lib.elem stdenv.hostPlatform.system lib.platforms.mesaPlatforms
+, withWebKit ? stdenv.isDarwin
+, webkitgtk
+, setfile
+, AGL
+, Carbon
+, Cocoa
+, Kernel
+, QTKit
+, AVFoundation
+, AVKit
+, WebKit
+}:
+
+assert withGtk2 -> (!withWebKit);
+
+let
+  gtk = if withGtk2 then gtk2 else gtk3;
+in
+stdenv.mkDerivation rec {
+  pname = "wxwidgets";
+  version = "3.2.1";
+
+  src = fetchFromGitHub {
+    owner = "wxWidgets";
+    repo = "wxWidgets";
+    rev = "v${version}";
+    hash = "sha256-wrrMR8McwJRIHTdmvaVP4DjumJMZ9Ik7WEUvD+tJCdM=";
+    fetchSubmodules = true;
```

Comment:
I don't want to vendor all those core libraries tbh https://github.com/wxWidgets/wxWidgets/blob/master/.gitmodules

Reply:
removed the fetchsubmodules. now it's downloading catch (I didn't managed to use the nixpkgs catch2) and nanosvg (it's not in nixpkgs) explicitly.